src/mbgl/renderer/painter.cpp

```diff
@@ -91,6 +84,17 @@ void Painter::render(const Style& style, const FrameData& frame_, View& view, Sp
     if (frame.contextMode == GLContextMode::Shared) {
         context.setDirtyState();
     }
+    
+    #ifndef NDEBUG
+        gl::debugging::enable();
+    #endif
+        shaderParameters = gl::ShaderParameters{ frame.pixelRatio, false };
+        shaders = std::make_unique<Shaders>(context, shaderParameters);
+    #ifndef NDEBUG
+        shaderParametersOverdraw = gl::ShaderParameters{ frame.pixelRatio, true };
+        overdrawShaders = std::make_unique<Shaders>(context, shaderParametersOverdraw);
```

Comment:
Why did you move this to the `render()` method? This means the shaders are initialized on every frame, which isn't efficient.